### PR TITLE
Migrate from GitLab CI to Travis CI

### PR DESCRIPTION
**Pull request by mpdeimos**
Open-sourcing the JaCoCo Agent requires some infrastructure changes like moving to Travis CI.
This is a rundown of the major changes:

* Use (default) Travis build step `assemble` instead of `dist` that also creates the shadow JAR
* Publish artifacts from Travis to GitLab Tag
* Use Flexmark instead of Pandoc for PDF generation
* Use Docker Cloud Build for building the Docker container (note: I don't think the container does anything useful. There isn't even Java installed :/)

Further considerations:
* We could use the version number of a tag instead of specifying it in the Gradle file.
* Do we want to upload old releases?
* With Travis we could automatically build against several jacoco versions. Do we want this?

### CODE REVIEW

**Comment by karottenreibe** [comment]
General comments:

- the Docker container is supposed to be mounted as a volume into a container running a JVM application. Thus it doesn't need to have Java installed. It's just an easy way to distribute the agent's Jar inside a Docker setup
- I wouldn't build against several JaCoCo versions. I see no benefit and a big drawback: if we ever need to work around issues in one version, we'll have to maintain that workaround and activate it conditionally. I'd rather have a fixed JaCoCo version that we update on demand and in a controlled process including testing

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+- rm -fr $HOME/.gradle/caches/*/fileHashes/
+- rm -fr $HOME/.gradle/caches/jars-3/*/buildSrc.jar
+cache:
+  directories:
+  - "$HOME/.gradle/caches/"
+  - "$HOME/.gradle/wrapper/"
+deploy:
+  provider: releases
+  api_key:
+    secure: iP2O3E7KrVlnw6E0acQ1iz2JEAwzAtMDUXbls14OlrZOz3Sc1Ti1UlF87F8FWaxpTGd2DnMue06XyO4XyTV4ijohWskxDL0BVOgKjqPtjLLB5rIxpVEW7BVM3jUc0gn7xAz2Ds2GI5gqDXBo7o5dGTB34KtaQYNjVejNOt3XfkKvcNSw+cHL9LU/lq0yAFmrKPkxDeUWmm8QmYzuXoGW/KiRcuNWe29Lg0bDoZ2j3DeA5qi8OyeZCa6I9t5rBhzDZa3RsA2XzsuxYGFHNL+jmZbQA62XAq6iP+g+JdpDojgWbzN7E/kbOS1MQcRoKX+3IyHKoOJbAS/nO439kg7s3qagJV+VkWRt2VrgkzWWn1vTUzE0u+NfadPhjtNiy0emNsqowGFdOiDVko0/OckQHDR1s9sxPt2akXcQJ82wPzg78ULdh+hJb/mr8SHcuxD5enV2mxRvxsRBMEJsluMOwg5q5k+EJkA0JOy5ZFF82z2+bA3nlma+Jcx6ymJBqez8SooYEekG7wR2lBHaQ17TamX5SI86k1yGK6hOSiHwmT8Fq50ZRF9kGcz2VR3Rp9WjcA6SN3Vs/t3pBxQUe/HDXzhW0U+aCxNLXmN+XG8IKdeNev0E5UDg1c1P2KVxap+gPxfQetBLuOfPl0KPGZ++bm/ZBkaQ5W2wxCwrUs8uuLQ=
```
Comment (karottenreibe) [inline .travis.yml]: did you check these credentials into version control on purpose?

File: LICENSE [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+      To apply the Apache License to your work, attach the following
+      boilerplate notice, with the fields enclosed by brackets "[]"
+      replaced with your own identifying information. (Don't include
+      the brackets!)  The text should be enclosed in the appropriate
+      comment syntax for the file format. We also recommend that a
+      file or class name and description of purpose be included on the
+      same "printed page" as the copyright notice for easier
+      identification within third-party archives.
+
+   Copyright [yyyy] [name of copyright owner]
```
Comment (karottenreibe) [inline LICENSE]: please replace placeholders

File: README.md [hunk 3]
```diff
@@ -1,35 +1,32 @@
...
+* Use pull requests. Complete the "Definition of Done" for every pull request.
+* There's a Teamscale project, please fix all findings before submitting your pull request for review. The Teamscale coding guidelines and Definition of Done apply as far as possible with the available tooling.
 
-All tags are built automatically. Only use builds from tagged commits. This ensures that
-we always know which code a customer is using.
+### Publishing
 
-The build server will build the project and you can download a build artifact with the dist afterwards.
-If the artifcats have been removed already, trigger the build again in GitLab.
+After merging, please create a GitHub Release tag with the version number, e.g. `v8.1.0`
```
Comment (karottenreibe) [inline README.md]: Please mention that we use semantic versioning and link to the semver website and our changelog

File: agent/build.gradle [hunk 4]
```diff
@@ -78,23 +82,13 @@ startShadowScripts {
...
-
 distributions {
 	shadow {
 		baseName = 'teamscale-jacoco-agent'
 		contents {
-			from(doc) {
-				include '**'
+			from(usageToPdf) {
 				into 'documentation'
+				rename 'USAGE.pdf', 'userguide.pdf'
```
Comment (karottenreibe) [inline agent/build.gradle]: we should rename USAGE.md to userguide.md and put it in a "documentation" folder. The name USAGE is a renmant of  the distant past :P

File: javaws-wrapper/build.gradle [hunk 5]
```diff
@@ -20,38 +24,21 @@ startShadowScripts {
...
-	commandLine 'pandoc', 'README.md', '-o', "$outputDir/README.pdf"
-}
-
 distributions {
 	shadow {
 		baseName = 'javaws'
 		contents {
-			from(doc) {
-				include '**'
+			from(readmeToPdf) {
```
Comment (karottenreibe) [inline javaws-wrapper/build.gradle]: we should name and place the documentation consistently with the agent. e.g. also under documentation/userguide.md

File: README.md [hunk 6]
```diff
@@ -1,35 +1,32 @@
...
+* [Teamscale JaCoCo Agent](agent/USAGE.md)
+* [Java Web Start Wrapper](javaws-wrapper/README.md)
 
-In case Gitlab has pruned the artifacts (happens after 1 week) you can simply re-run the
-build.
+## Development
 
-__Use only zips produced by builds for a version tag (Tags tab in Gitlab). This ensures
-that we always know which exact version a customer has deployed.__
+### Contributing
```
Comment (karottenreibe) [inline README.md]: Please mention how to build locally

File: agent/build.gradle (hunk 4 above)
Comment (karottenreibe) [reply agent/build.gradle]: you can check the fileupload-service repo for how to do that

File: .travis.yml (hunk 1 above)
Comment (mpdeimos) [reply .travis.yml]: These are the encrypted credentials...

File: LICENSE (hunk 2 above)
Comment (mpdeimos) [reply LICENSE]: Removed section

File: javaws-wrapper/build.gradle (hunk 5 above)
Comment (mpdeimos) [reply javaws-wrapper/build.gradle]: i've renamed all documentation files now README.md (this way GitHub shows the file when opening a subfolder. On build these are renamed to userguide.

**Comment by mpdeimos** [comment]
Done w/ re-work and merging. conflicts were solely regarding tabs vs spaces in gradle files. imho our coding guidelines prefer tabs, so we should use these here as well.

**Pull request merged**